doc/releases/release-notes-3.7.rst

```diff
@@ -66,7 +66,13 @@ Deprecated in this release
   * In efforts to use Kconfig options that map directly to the Options and Option Groups in
     IEEE 1003.1-2017, the following Kconfig options have been deprecated (replaced by):
 
+    * :kconfig:option:`CONFIG_MAX_TIMER_COUNT` (:kconfig:option:`CONFIG_POSIX_TIMER_MAX`)
```

Comment:
I suggest to merge all deprecation documentation commits to the actual commit that does the deprecation.

Reply:
I'm not crazy about having release notes commits mixed with code commits.

I could squash all of the release notes to 1 message though.

Does that work?

Reply:
Fixed!